code/src/main/java/com/onlinecontacttracing/servlets/messaging/MessageSendingServlet.java

```diff
@@ -1,36 +1,75 @@
 import com.onlinecontacttracing.storage.PersonEmail;
-import com.onlinecontacttracing.storage.NotificationBatch;
-import java.util.Collection;
 import java.util.Collections;
-import com.google.api.client.http.HttpTransport;
-import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
-import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
-import javax.servlet.http.HttpServletResponse;
-import javax.servlet.http.HttpServletRequest;
-import com.google.api.client.json.JsonFactory;
-import java.util.logging.Logger;
-import javax.servlet.annotation.WebServlet;
-import javax.servlet.http.HttpServlet;
 import java.security.GeneralSecurityException;
+import com.onlinecontacttracing.storage.NotificationBatch;
+import static com.googlecode.objectify.ObjectifyService.ofy;
 
-@WebServlet("/message-sender")
-public class MessageSender extends HttpServlet {
 
+@WebServlet("/send-messages")
+
+/**
+* This servlet will send out messages to all the user's approved contacts.
+*/
+public class MessageSendingServlet extends HttpServlet {
+
+  static final Logger log = Logger.getLogger(DeleteNegativeUserLocationsServlet.class.getName());
   private static final String CLIENT_ID = "83357506440-etvnksinbmnpj8eji6dk5ss0tbk9fq4g.apps.googleusercontent.com";
   private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
-  private static final String CLIENT_ID = "83357506440-etvnksinbmnpj8eji6dk5ss0tbk9fq4g.apps.googleusercontent.com";
 
-  static final Logger log = Logger.getLogger(MessageSender.class.getName());
-    
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String idTokenString = request.getParameter("idToken");
+    String systemMessageName = "VERSION_1";
+    String localityResourceName = "US";
+    String messageLanguage = "SP";
+
+    SystemMessage systemMessage = SystemMessage.getSystemMessageFromString(systemMessageName);
+    LocalityResource localityResource = LocalityResource.getLocalityResourceFromString(localityResourceName);
+
+    try {
+      NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+
+      // Make verifier to get payload
+      GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier.Builder(httpTransport, JSON_FACTORY)
+        .setAudience(Collections.singletonList(CLIENT_ID))
+        .build();
+      GoogleIdToken idToken = verifier.verify(idTokenString);
+      Payload payload = idToken.getPayload();
+      String userId = payload.getSubject();
+
+      PositiveUser positiveUser = ofy().load().type(PositiveUser.class).id(userId).now();
+      
+      String message = "custom message will be retrieved from params";
+      CompiledMessage compiledMessage = new CompiledMessage(systemMessage, localityResource, message, positiveUser);
+
+      EmailSender.sendEmailsOut("COVID-19 Updates", compiledMessage, messageLanguage);
```

Comment:
plz make this a constant, and also maybe make it more descriptive?

Example: COVID-19 Potential Exposure Alert

Reply:
Also, actually this doesn't work because the title should change based on the language specified. Can you make EmailSubject a class that implements HasEnglishTranslation/HasSpanishTranslation?